board/fancom/hmi/hmi.c

```diff
@@ -0,0 +1,514 @@
+
+	ret = bcm2835_mbox_call_prop(BCM2835_MBOX_PROP_CHAN, &msg->hdr);
+	if (ret) {
+		printf("bcm2835: Could not query board serial\n");
+		/* Ignore error; not critical */
+		return;
+	}
+
+	snprintf(serial_string, sizeof(serial_string), "%016llx",
+		 msg->get_board_serial.body.resp.serial);
+	env_set("serial#", serial_string);
+}
+
+int misc_init_r(void)
+{
+	set_fdt_addr();
+	//set_fdtfile();
+	//set_usbethaddr();
+#ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
+	set_board_info();
+#endif
+	//set_serial_number();
+
+	return 0;
+}
+
+static void get_board_rev(void)
+{
+	ALLOC_CACHE_ALIGN_BUFFER(struct msg_get_board_rev, msg, 1);
+	int ret;
+	const struct rpi_model *models;
+	uint32_t models_count;
+
+	BCM2835_MBOX_INIT_HDR(msg);
+	BCM2835_MBOX_INIT_TAG(&msg->get_board_rev, GET_BOARD_REV);
+
+	ret = bcm2835_mbox_call_prop(BCM2835_MBOX_PROP_CHAN, &msg->hdr);
+	if (ret) {
+		printf("bcm2835: Could not query board revision\n");
+		/* Ignore error; not critical */
+		return;
+	}
+
+	/*
+	 * For details of old-vs-new scheme, see:
+	 * https://github.com/pimoroni/RPi.version/blob/master/RPi/version.py
+	 * http://www.raspberrypi.org/forums/viewtopic.php?f=63&t=99293&p=690282
+	 * (a few posts down)
+	 *
+	 * For the RPi 1, bit 24 is the "warranty bit", so we mask off just the
+	 * lower byte to use as the board rev:
+	 * http://www.raspberrypi.org/forums/viewtopic.php?f=63&t=98367&start=250
+	 * http://www.raspberrypi.org/forums/viewtopic.php?f=31&t=20594
+	 */
+	revision = msg->get_board_rev.body.resp.rev;
+	if (revision & 0x800000) {
+		rev_scheme = 1;
+		rev_type = (revision >> 4) & 0xff;
+		models = rpi_models_new_scheme;
+		models_count = ARRAY_SIZE(rpi_models_new_scheme);
+	} else {
+		rev_scheme = 0;
+		rev_type = revision & 0xff;
+		models = rpi_models_old_scheme;
+		models_count = ARRAY_SIZE(rpi_models_old_scheme);
+	}
+	if (rev_type >= models_count) {
+		printf("RPI: Board rev 0x%x outside known range\n", rev_type);
+		model = &rpi_model_unknown;
+	} else if (!models[rev_type].name) {
+		printf("RPI: Board rev 0x%x unknown\n", rev_type);
+		model = &rpi_model_unknown;
+	} else {
+		model = &models[rev_type];
+	}
+
+	printf("RPI %s (0x%x)\n", model->name, revision);
+}
+
+int board_init(void)
+{
+#ifdef CONFIG_HW_WATCHDOG
+	hw_watchdog_init();
+#endif
+
+	get_board_rev();
+
+	gd->bd->bi_boot_params = 0x100;
+
+	return bcm2835_power_on_module(BCM2835_MBOX_POWER_DEVID_USB_HCD);
+}
+
+/*
+ * If the firmware passed a device tree use it for U-Boot.
+ */
+void *board_fdt_blob_setup(void)
+{
+	if (fdt_magic(fw_dtb_pointer) != FDT_MAGIC)
+		return NULL;
+	return (void *)fw_dtb_pointer;
+}
+
+int ft_board_setup(void *blob, struct bd_info *bd)
+{
+//	/*
+//	 * For now, we simply always add the simplefb DT node. Later, we
+//	 * should be more intelligent, and e.g. only do this if no enabled DT
+//	 * node exists for the "real" graphics driver.
+//	 */
+//	lcd_dt_simplefb_add_node(blob);
```

Comment:
Why has this been commented out?, then the CONFIG_RPI_EFI_NR_SPIN_PAGES doesn't make sense anymore in the Kconfig

Reply:
This was part of a test to get blue/green working. I will reenable and test it.